src/firebolt/utils/usage_tracker.py

```diff
@@ -0,0 +1,159 @@
+import inspect
+import logging
+from enum import Enum
+from platform import python_version, release, system
+from sys import modules
+from typing import Dict, List, Optional, Tuple, Union
+
+from firebolt import __version__
+
+logger = logging.getLogger(__name__)
+
+
+class Format(Enum):
```

Comment:
Why do we need this type? I believe if we'll switch to another tracking method, we'll just rewrite the logic

Reply:
I'm not sure if we'll get rid of the user-agent completely, but sure, let's leave that for the time when decision needs to be made.